File: Problems/0053.Maximum-Subarray/MaximumSubarray_test.go
package leetcode

import (
	"testing"
)

func Test(t *testing.T) {
	testSuits := []struct {
		name string
		nums []int
		ans  int
	}{
		{"Example 1", []int{-2, 1, -3, 4, -1, 2, 1, -5, 4}, 6},
		{"Example 2", []int{-1}, -1},
	}

	for _, test := range testSuits {
		got := maxSubArray(test.nums)

		if got != test.ans {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, got)
		}
	}
}

File: leetcode/0053.Maximum-Subarray/MaximumSubarray_test.py
from MaximumSubarray import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
        ("Example-1", [-2,1,-3,4,-1,2,1,-5,4], 6),
        ("Example-2", [-1], -1),
]

@pytest.mark.parametrize(
    "example, nums, ans", 
    deepcopy(TEST_SUITS)
)
def test(example, nums, ans):
    assert SOL.maxSubArray(nums) == ans